DSCResources/MSFT_xDnsRecord/MSFT_xDnsRecord.psm1

```diff
@@ -36,20 +36,24 @@ function Get-TargetResource
         [System.String]
         $Target,
 
+        [System.String]
+        $DnsServer = "localhost",
+
         [ValidateSet('Present','Absent')]
         [System.String]
         $Ensure = 'Present'
     )
 
-    Write-Verbose -Message ($LocalizedData.GettingDnsRecordMessage -f $Name, $Type, $Zone)
-    $record = Get-DnsServerResourceRecord -ZoneName $Zone -Name $Name -ErrorAction SilentlyContinue
+    Write-Verbose -Message ($LocalizedData.GettingDnsRecordMessage -f $Name, $Type, $Zone, $DnsServer)
+    $record = Get-DnsServerResourceRecord -ZoneName $Zone -Name $Name -ComputerName $DnsServer -ErrorAction SilentlyContinue
     
-    if ($record -eq $null) 
+    if ([bool]($record -eq $null))
```

Comment:
The cast to a boolean isn't needed here.
Also, this won't pass PSScriptAnalyzer. The `$null` should be on the left side of the comparison instead of the right.
You can read more about it here: https://github.com/PowerShell/PSScriptAnalyzer/blob/development/RuleDocumentation/PossibleIncorrectComparisonWithNull.md